source/non_matching/immersed_surface_quadrature.cc

```diff
@@ -28,7 +28,7 @@ namespace NonMatching
   {
     AssertDimension(weights.size(), points.size());
     AssertDimension(normals.size(), points.size());
-    for (auto normal : normals)
+    for (const auto normal : normals)
```

Comment:
don't we usually use `const auto &`?

Reply:
Yes, we do and clang-tidy also does that for large enough variables. The configuration I used for running `clang-tidy` should do that for variables larger than 16 bytes. If necessary, I can try to find out how to change this value...

Reply:
I see. I don't care too much about it here, but we pass `Tensor<1,dim>` objects by reference, so I thought we should do the same here. Your choice.

Reply:
I agree to pass by reference here. `Wloop-analysis` catches this so that should not be an issue for the upcoming PRs.